ironfish/src/wallet/wallet.ts

```diff
@@ -739,6 +739,49 @@ export class Wallet {
     return transaction
   }
 
+  async burn(
+    memPool: MemPool,
+    account: Account,
+    assetIdentifier: Buffer,
+    value: bigint,
+    fee: bigint,
+    transactionExpirationDelta: number,
+    expiration?: number,
+  ): Promise<Transaction> {
+    const heaviestHead = this.chain.head
+    if (heaviestHead === null) {
+      throw new Error('You must have a genesis block to create a transaction')
+    }
+
+    expiration = expiration ?? heaviestHead.sequence + transactionExpirationDelta
+    if (this.chain.verifier.isExpiredSequence(expiration, this.chain.head.sequence)) {
+      throw new Error('Invalid expiration sequence for transaction')
+    }
+
```

Comment:
Should we do any validation of assetIdentifier here, or is it done inside of `createTransaction`?

Reply:
Creating and posting the transaction will validate the sender owns note(s) [+ correct witness path] with adequate balance to burn the provided value for the given asset identifier